Plugins/Public/base_plugin/PlayerCommands.cpp

```diff
@@ -1810,6 +1810,29 @@ namespace PlayerCommands
 			}
 			PrintUserCmdText(client, L"ERR Commodity does not exist");
 		}
+		else if (cmd == L"public" || cmd == L"private")
+		{
+			int item = ToInt(GetParam(args, ' ', 2));
+
+			int curr_item = 1;
+			for (map<UINT, MARKET_ITEM>::iterator i = base->market_items.begin(); i != base->market_items.end(); ++i, curr_item++)
+			{
+				if (curr_item == item)

```

Comment:
can we do some trick like `map<UINT, MARKET_ITEM>::iterator i = base->market_items.begin() + item - 1` (with a bounds check) or something to achieve the same result as this loop?

Reply:
I think so

Reply:
Yeah I was able to do that